system/src/system.cpp

```diff
@@ -66,6 +69,52 @@ int lua_threadcount(lua_State* L)
     return 1;
 }
 
+int lua_freememory(lua_State* L)
+{
+    lua_pushinteger(L, uv_get_free_memory()); // FIXME: this is a 64 bit int, deal with it?
+
+    return 1;
+}
+
+int lua_totalmemory(lua_State* L)
+{
+    lua_pushinteger(L, uv_get_total_memory()); // FIXME: this is a 64 bit int, deal with it?
+
+    return 1;
+}
+
+int lua_hostname(lua_State* L)
+{
+    size_t sz = UV_MAXHOSTNAMESIZE;
+    char* hostname = (char*) malloc(sz);
+
+    int res = uv_os_gethostname(hostname, &sz);
+    if (res != 0) {
+        lua_pushfstringL(L, "libuv error: %s", uv_strerror(res));
```

Comment:
You should properly handle this error, and not just use UV_MAXHOSTNAMESIZE.

You could look at the implementation of `process.cwd` for this - it does it properly.